### PR TITLE
Introduce import layout configuration for ImportOrderingRule

### LINKED ISSUE #527: Alphabetical import ordering enabled by default is a problem in Android Studio/IntelliJ
In our project `ktlint 0.34.0` now reports a lot of incorrect import ordering, although we use a default import layou of IDE. All our developers have "Optimize imports" enabled by default, but this still creates an order like the following (which seems to be default one of Kotlin IDE plugin):

```
import ru.example.a
import ru.example.b
import ru.example.c
import javax.inject.Inject
```

Notice that the last one comes in non alphabetical order. 
The unfortunate thing is that this cannot be changed.

Import order editing is there only for Java CodeStyle, kotlin plugin seems not to provide it. Correct me if I'm wrong.

If you think that this is still something that should be default, then please close.

But currently I suspect that many Android Studio/IntelliJ users will be disabling this rule, otherwise they'll have an endless conflict between IDE autoformatter/ktlint.

### PR DESCRIPTION
**Pull request by romtsn**
Fix #527

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -13,6 +13,9 @@ insert_final_newline = true
 [*.{java,kt,kts,scala,rs,xml,kt.spec,kts.spec}]
 indent_size = 4
 
+[*.{kt,kts}]
+kotlin_imports_layout=ascii
```
Comment (romtsn) [inline .editorconfig:17]: I'm not sure if we should use "idea" for ktlint itself.. If yes, then I'd prefer to do it in a separate PR afterwards, but can also do it here

File: ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt [hunk 2]
```diff
@@ -1,53 +1,171 @@
...
+    }
+
     override fun visit(
         node: ASTNode,
         autoCorrect: Boolean,
         emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
     ) {
+        if (node.isRoot()) {
+            val editorConfig = node.getUserData(KtLint.EDITOR_CONFIG_USER_DATA_KEY)!!
+            importsLayout = editorConfig.get("kotlin_imports_layout")
```
Comment (romtsn) [inline ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt:76]: I've decided to not put `ij_` in front, because it would look weird a bit if it was `ij_kotlin_imports_layout=idea`. But open for suggestions

File: .editorconfig (hunk 1 above)
Comment (shashachu) [reply .editorconfig:17]: I personally use IDEA but I rarely use the IDE's code formatting so I don't have a strong opinion.

File: ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt [hunk 3]
```diff
@@ -1,53 +1,171 @@
...
+                ?: "idea" // default to idea in case there's no such entry in .editorconfig
```
Comment (gabrielittner) [inline ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt:77]: For Android projects this default violates the style guide https://developer.android.com/kotlin/style-guide#import_statements

Comment (romtsn) [reply ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt:77]: It violates the style guide, but it complies with how IDEA/Android Studio actually sorts your imports, so I'd say it's a good middle-ground. You can always configure it to "ascii" if you absolutely want to comply with Android style guide.

Also I don't really see a point in defining a style guide, such as you cannot set up Android Studio to comply with it, because there's literally no such capability for Kotlin yet.

File: .editorconfig (hunk 1 above)
Comment (Tapchicoma) [reply .editorconfig:17]: I would say it is fine. And this property could be in the future not really IDEA specific.

File: README.md [hunk 4]
```diff
@@ -82,6 +82,17 @@ max_line_length=off
 # Note that rules in any ruleset other than the standard ruleset will need to be prefixed 
 # by the ruleset identifier.
 disabled_rules=no-wildcard-imports,experimental:annotation,my-custom-ruleset:my-custom-rule
+
+# Defines the imports layout. There are predefined layouts like "ascii" or "idea", as well as a custom layout.
```
Comment (Tapchicoma) [inline README.md:86]: Nice documentation :+1:. Probably should mention that this will be available starting Kotlin 1.4-M3 release.

File: ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt (hunk 2 above)
Comment (Tapchicoma) [reply ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt:76]: Hm, I think anyway it should have `ij_` prefix in the `.editorconfig` so IDEA will pickup for codestyle? Correct me if I am wrong.

If yet, I would prefer using it here as well. Could rethink this approach in the future.

File: ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt (hunk 3 above)
Comment (Tapchicoma) [reply ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt:77]: Probably you could reuse `ANDROID_USER_DATA_KEY` to select default config `ascii` for android projects.

File: ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/internal/importordering/ImportLayoutParser.kt [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            var hasAlias = false
+            var withSubpackages = false
+            if (import.startsWith(ALIAS_CHAR)) {
+                import = import.substring(1).trim()
+                hasAlias = true
+            }
+            if (import.endsWith(WILDCARD_CHAR)) {
+                withSubpackages = true
+            }
+            imports += if (import == WILDCARD_CHAR) {
```
Comment (Tapchicoma) [inline ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/internal/importordering/ImportLayoutParser.kt:36]: Nit: probably you could use transformations like `fold` on `importsList` to transform it to end result instead of using `forEach`

Comment (romtsn) [reply ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/internal/importordering/ImportLayoutParser.kt:36]: True, it was a bit different in the initial implementation, but now I think it's doable via transformation operators.. probabl `map` fits best here, will change that 👍

File: README.md (hunk 4 above)
Comment (romtsn) [reply README.md:86]: But I mean it's available for ktlint already now, so you can specify the custom layout for ktlint, there's just no such capability in Kotlin IDEA plugin yet.. or do you want to also mention about the kotlin plugin just in case?

File: ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt (hunk 3 above)
Comment (romtsn) [reply ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt:77]: I could certainly do that, but I think this will result in 95% of android devs to set it back to `idea`, until Kotlin 1.4 is officially out (I guess only a small percentage is using milestone versions), otherwise it will lead to the same conflicts and frustration as it is now.. I'd rather change it after 1.4 is officially there - what do you guys think?

Comment (romtsn) [reply ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt]: Implemented default to `ascii` in case of android flag enabled

File: ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt (hunk 2 above)
Comment (romtsn) [reply ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt]: As discussed over slack, added support for both names

File: README.md (hunk 4 above)
Comment (Tapchicoma) [reply README.md:86]: You are right, we could mention it later when it will be publicly available.
